addons/l10n_it_edi/data/invoice_it_template.xml

```diff
@@ -128,10 +134,32 @@
                 <Divisa t-esc="currency.name"/>
                 <Data t-esc="format_date(record.invoice_date)"/>
                 <Numero t-esc="format_alphanumeric(record.name[-20:])"/>
+                <t t-if="withholding_values">
+                    <t t-foreach="withholding_values" t-as="withholding">
+                        <DatiRitenuta>
+                            <TipoRitenuta t-esc="format_alphanumeric(withholding['type'])"/>
+                            <ImportoRitenuta t-esc="format_monetary(withholding['amount'], currency)"/>
+                            <AliquotaRitenuta t-esc="format_numbers(withholding['percentage'])"/>
+                            <CausalePagamento t-esc="format_alphanumeric(withholding['reason'])"/>
+                        </DatiRitenuta>
+                    </t>
+                </t>
                 <DatiBollo t-if="record.l10n_it_stamp_duty">
                     <BolloVirtuale>SI</BolloVirtuale>
                     <ImportoBollo t-esc="format_numbers(record.l10n_it_stamp_duty)"/>
                 </DatiBollo>
+                <t t-if="pension_fund_values" t-foreach="pension_fund_values" t-as="pension_fund">
+                <DatiCassaPrevidenziale t-if="pension_fund">
+                    <TipoCassa t-if="pension_fund.get('tipocassa')" t-esc="format_alphanumeric(pension_fund['tipocassa'])"/>
+                    <AlCassa t-if="pension_fund.get('alcassa')" t-esc="format_numbers(pension_fund['alcassa'])"/>
+                    <ImportoContributoCassa t-if="pension_fund.get('importocontributocassa')" t-esc="format_monetary(pension_fund['importocontributocassa'], currency)"/>
+                    <ImponibileCassa t-if="pension_fund.get('imponibilecassa')" t-esc="format_monetary(pension_fund['imponibilecassa'], currency)"/>
```

Comment:
these values are all optional with the .get?

Reply:
Hmmm... yes, this is also wrong btw, as the Aliquota could be 0.00 which is false